Yoast/wordpress-seo
# Remove settings introduction modal

## Context


* We want to remove the introductory modal for the settings page.

## Summary


This PR can be summarized in the following changelog entry:

* Removes the introductory modal for the settings page.

## Relevant technical choices:

* An upgrade routine has been added to remove the introuction-related option in `wp_usermeta`.

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

**Preliminari steps**
* Do not apply this PR
* Create a new user with `SEO manager` as role
* Login in the backend with the newly created user
  * Go to `Yoast SEO` -> `Settings` 
  * Dismiss the introduction modal
 
**Test the modal removal**
* Apply this PR
* Create a new user
* Login in the backend with the newly created user
  * Go to `Yoast SEO` -> `Settings` 
  * verify no introduction modal is visualized

**Test the upgrade routine** 
* Make sure you have the `Yoast Test Helper` installed and activated
* With a MySQL client perform the following query:
```
SELECT * FROM wp_usermeta WHERE meta_key="_yoast_settings_introduction";
```
( it should return more than one result)
*  Go to `Tools` - > `Yoast Test`
*  Set DB Version for Yoast SEO to 20.4 and save
* Perform the same query as above and verify it returns no results

* Perform also a smoke test on the Settings page to make sure it still works and renders properly

#### Relevant test scenarios


### Test instructions for QA when the code is in the RC



## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

* N/A

## UI changes


## Other environments


## Documentation


## Quality assurance


## Innovation


Fixes https://github.com/Yoast/wordpress-seo/issues/20113
